code/controllers/subsystems/ai.dm

```diff
@@ -26,6 +26,8 @@ SUBSYSTEM_DEF(ai)
 		--currentrun.len
 		if(!A || QDELETED(A) || A.busy) // Doesn't exist or won't exist soon or not doing it this tick
 			continue
+		if(A.holder.client && !A.autopilot)
+			continue
```

Comment:
This triple-dereference is probably not a great idea for performance (same with SSaifast).

It would probably be better to do something like...
- Mob login() checks if there's an AI holder
- If there's an AI holder, and !holder.autopilot, it removes the holder from both AI subsystems with `manage_processing(0)`
- Mob logout() checks if there's an AI holder
- If there's an AI holder, cheat a bit and call holder_stat_change on it, pass it `src, stat, stat`, that should readd it to the subsystem BUT ONLY IF it's still alive, which is how it should be handled for mobs without clients or anything

Reply:
I tried the manage_processing thing with my first attempt before going for this approach and it didn't work.

Reply:
One note: This has the downside of giving AI control back to mobs if you're an admin and aghost, because your client leaves the mob (though your solution also has that problem).